src/Stratis.Bitcoin.Features.BlockStore/BlockStoreFeature.cs

```diff
@@ -141,6 +169,7 @@ public static IFullNodeBuilder UseBlockStore(this IFullNodeBuilder fullNodeBuild
                     {
                         services.AddSingleton<IBlockStoreQueue, BlockStoreQueue>().AddSingleton<IBlockStore>(provider => provider.GetService<IBlockStoreQueue>());
                         services.AddSingleton<IBlockRepository, BlockRepository>();
+                        services.AddSingleton<IPrunedBlockRepository, PrunedBlockRepository>();
```

Comment:
Shouldn't it be the same instance as block repo? 

So instead of block repo you have pruned block repo, but you have 2 interfaces. So when you ask for block repo or pruned block repo you always receive instance of pruned block repo

Reply:
Block Repository stands on its own. Block repo is injected into PrunedBlockRepo.

This is too big a change to make now and can be improved later 👍